tool/lib/commands/generate_changelog.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show jsonDecode;
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+
+import 'package:args/command_runner.dart';
+
+import '../model.dart';
+
+class GenerateChangelogCommand extends Command {
+  // You can authorize your access if you run into a github rate limit.
+  // Don't check in your passwords or auth tokens.
+  static const auth = '';
+  @override
+  String get name => 'generate-changelog';
+
+  @override
+  String get description =>
+      'Generates a changelog of all commits since a given tag.';
+
+  @override
+  Future run() async {
+    final repo = DevToolsRepo.getInstance();
+    String devtoolsVersionFile =
+        await File('${repo.repoPath}/packages/devtools_app/lib/devtools.dart')
+            .readAsString();
+    const versionDeclarationPrefix = 'const String version = \'';
+    final versionDeclaration =
+        devtoolsVersionFile.indexOf(versionDeclarationPrefix);
+    final versionEnd = devtoolsVersionFile.indexOf('\';');
+    final version = devtoolsVersionFile.substring(
+        versionDeclaration + versionDeclarationPrefix.length, versionEnd);
+    final List tags = jsonDecode((await http.get(
+            Uri.https('${auth}api.github.com', '/repos/flutter/devtools/tags')))
+        .body);
+
+    print('Current Devtools version is $version.  Retrieving the tagged commit '
+        'with the closest version number to this version.');
+    var closestTag = tags.first;
+    for (var tag in tags) {
+      if (tag['name'].split('-').length > 1) {
+        // This was a dev build.
+        continue;
+      }
+      if (getVersion(tag['name']) < getVersion(version) &&
+          (closestTag == null ||
+              getVersion(tag['name']) > getVersion(closestTag['name']))) {
+        print('it is the closest tag');
+        closestTag = tag;
+      }
+    }
+
+    print('getting the date of the tagged commit for ${closestTag["name"]}.');
```

Comment:
you should be able to use single quotes here since they'd be inside { }. Same comment for uses of " elsewhere

Reply:
TIL, thanks!